cmd/verify/network/cmd_test.go

```diff
@@ -0,0 +1,488 @@
+		Expect(err).ToNot(BeNil())
+		Expect(err.Error()).To(
+			ContainSubstring("One and only one of cluster or subnet-ids is required"))
+	})
+	It("Fails if no --role-arn without --cluster", func() {
+		cmd.Flags().Set(subnetIDsFlag, "abc,def")
+		err := runWithRuntime(r, cmd)
+		Expect(err).ToNot(BeNil())
+		Expect(err.Error()).To(
+			ContainSubstring("role-arn is required"))
+	})
+	DescribeTable("Test --cluster with various statuses",
+		func(state string, expected string) {
+			cmd.Flags().Lookup(statusOnlyFlag).Changed = true
+			cmd.Flags().Set(clusterFlag, "tomckay-vpc")
+
+			var clusterJson map[string]any
+			json.Unmarshal([]byte(clustersSuccess), &clusterJson)
+			clusterJson["items"].([]interface{})[0].(map[string]interface{})["state"] = state
+			clusterJson["items"].([]interface{})[0].(map[string]interface{})["status"].(map[string]interface{})["state"] = state
+			updatedCluster, _ := json.Marshal(clusterJson)
+
+			// GET /api/clusters_mgmt/v1/clusters
+			apiServer.AppendHandlers(
+				RespondWithJSON(
+					http.StatusOK,
+					string(updatedCluster),
+				),
+			)
+			// GET /api/clusters_mgmt/v1/network_verifications/subnetA
+			apiServer.AppendHandlers(
+				RespondWithJSON(
+					http.StatusOK,
+					subnetASuccess,
+				),
+			)
+			// GET /api/clusters_mgmt/v1/network_verifications/subnetB
+			apiServer.AppendHandlers(
+				RespondWithJSON(
+					http.StatusOK,
+					subnetBSuccess,
+				),
+			)
+			stdout, stderr, err := captureStdout(r, cmd)
+			Expect(err).To(BeNil())
+			Expect(stderr).To(Equal(""))
+			Expect(stdout).To(Equal(expected))
+		},
+		Entry("ready state", "ready", successOutputPendingComplete),
+		Entry("error state", "error", successOutputPendingComplete),
+		// TODO: add other states when upcoming PR to remove restrictions is merged
+		// Entry("hibernating state", "hibernating", successOutputPendingComplete),
+	)
+	It("Succeeds if --cluster with --role-arn", func() {
+		// GET /api/clusters_mgmt/v1/clusters
+		apiServer.AppendHandlers(
+			RespondWithJSON(
+				http.StatusOK,
+				clustersSuccess,
+			),
+		)
+		// POST /api/clusters_mgmt/v1/network_verifications
+		apiServer.AppendHandlers(
+			RespondWithJSON(
+				http.StatusOK,
+				subnetsSuccess,
+			),
+		)
+		// GET /api/clusters_mgmt/v1/network_verifications/subnetA
+		apiServer.AppendHandlers(
+			RespondWithJSON(
+				http.StatusOK,
+				subnetASuccess,
+			),
+		)
+		// GET /api/clusters_mgmt/v1/network_verifications/subnetB
+		apiServer.AppendHandlers(
+			RespondWithJSON(
+				http.StatusOK,
+				subnetBSuccess,
+			),
+		)
+		cmd.Flags().Set(clusterFlag, "tomckay-vpc")
+		cmd.Flags().Set(roleArnFlag, "arn:aws:iam::765374464689:role/tomckay-Installer-Role")
+		stdout, stderr, err := captureStdout(r, cmd)
+		Expect(err).To(BeNil())
+		Expect(stderr).To(Equal(""))
+		Expect(stdout).To(Equal(successOutputPendingComplete))
+	})
+})
+
+func captureStdout(runtime *rosa.Runtime, cmd *cobra.Command) (string, string, error) {
```

Comment:
nice, so we can also compare output. I suggest to move this outside the specific package, so it can be reused, as it is not specific to this use case

Reply:
agreed, perhaps `pkg/helpers/tests/helpers.go` (i'm open to alternative)

Reply:
`pkg/test/helpers.go` ? close to what we have on the backend